### PR TITLE
Adding stemflow

### PR DESCRIPTION
**Pull request by chenyangkang**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by chenyangkang** [comment]
@conda-forge/help-python, ready for review!

File: recipes/stemflow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 194af6f1247875eb7e0eadbec327f29e29d569484337cc45e003b7a565315814
+
+build:
+  number: 0
+  skip: true  # [win, osx]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
```
Comment (synapticarbors) [inline recipes/stemflow/meta.yaml:18]: `host` typically only requires `pip`, `python`, `setuptools` and `setuptools_scm`.

File: recipes/stemflow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 194af6f1247875eb7e0eadbec327f29e29d569484337cc45e003b7a565315814
+
+build:
+  number: 0
+  skip: true  # [win, osx]
```
Comment (synapticarbors) [inline recipes/stemflow/meta.yaml:14]: why are you skipping win and osx? If that's not necessary, then this should probably be `noarch: python`:
https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

File: recipes/stemflow/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    - matplotlib-base >=3.7.1
+    - numpy >=1.24.3
+    - pandas >=2.0.3
+    - python
+    - scikit-learn >=1.2.2
+    - seaborn >=0.11.2
+    - tqdm >=4.65.0
+    - xgboost >=2.0.0
+    - watermark >=2.4.3
+test:
```
Comment (synapticarbors) [inline recipes/stemflow/meta.yaml:45]: probably a good idea to add a `pip check` test:
https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L61-L64

File: recipes/stemflow/meta.yaml (hunk 1 above)
Comment (synapticarbors) [reply recipes/stemflow/meta.yaml:18]: Also, I don't see `setuptools_scm` used anywhere in this project. You likely only need `python` and `pip`.

File: recipes/stemflow/meta.yaml (hunk 2 above)
Comment (chenyangkang) [reply recipes/stemflow/meta.yaml]: Thanks for the hint. Fixed.

File: recipes/stemflow/meta.yaml (hunk 3 above)
Comment (chenyangkang) [reply recipes/stemflow/meta.yaml:34]: Thanks for the comment. `pip check` added.

File: recipes/stemflow/meta.yaml (hunk 1 above)
Comment (chenyangkang) [reply recipes/stemflow/meta.yaml:18]: Changed host to 
```
  host:
    - pip
    - python >=3.8
```

**Comment by chenyangkang** [comment]
@synapticarbors Hi I'm pretty new to conda-forge and sorry if this is a stupid mistake. I could not figure out why osx and win failed to build. Could you please provide any hint? Thanks a lot!

Yangkang

**Comment by synapticarbors** [comment]
xgboost 2.0.0 looks like it was just released in the last few days, and it doesn't look like we have it on conda-forge yet.

**Comment by jaimergp** [comment]
Rerunning this to see if Linux properly fails now. Sorry for the noise!

**Comment by synapticarbors** [comment]
It looks like the `host` section will require all of the run time dependencies since `setup.py` imports the package. It's only doing so to get the version, but that triggers the full import of everything in `__init__.py`. It looks like that was a recent change, where before the version was hardcoded.

**Comment by synapticarbors** [comment]
The failure on windows is ok as it's just complaining about the xgboost being missing. Since this is `noarch:python` we typically just ignore that and leave it to the user to obtain xgboost if they want it.

**Comment by chenyangkang** [comment]
> The failure on windows is ok as it's just complaining about the xgboost being missing. Since this is `noarch:python` we typically just ignore that and leave it to the user to obtain xgboost if they want it.

Thanks! I'm uploading a new version and I think it's ready to go now.

Do I have to go through all these recipe process each time I release a new version?

**Comment by synapticarbors** [comment]
Once the recipe is stable and working there shouldn't be much need for lots of changes. Additionally conda-forge has a bot that automatically opens PRs for updates that does the basic stuff like bumping the version and updating the hash. You typically just need to make sure the CI all passes and that the requirements are still correct.

**Comment by chenyangkang** [comment]
> Once the recipe is stable and working there shouldn't be much need for lots of changes. Additionally conda-forge has a bot that automatically opens PRs for updates that does the basic stuff like bumping the version and updating the hash. You typically just need to make sure the CI all passes and that the requirements are still correct.

Thanks for that @synapticarbors! Ready for merge now.

**Pull request merged**